packages/fiori-annotation-api/src/cds/adapter.ts

```diff
@@ -281,31 +286,36 @@ export class CDSAnnotationServiceAdapter implements AnnotationServiceAdapter, Ch
         const workspaceChanges: { [uri: string]: TextEdit[] } = {};
         this.clearState();
         const writers = new Map<string, CDSWriter>();
-        for (const change of changes) {
-            let writer = writers.get(change.uri);
-            const document = this.documents.get(change.uri);
-            const cachedFile = this.fileCache?.get(change.uri);
-            if (!document || cachedFile === undefined || !this.facade) {
-                continue;
-            }
-            if (!writer) {
-                //writable cds document (augment)
-                const textDocument = TextDocument.create(change.uri, 'cds', 0, cachedFile);
-                writer = new CDSWriter(
-                    this.facade,
-                    this.vocabularyService,
-                    document.ast,
-                    document.comments,
-                    document.tokens,
-                    textDocument,
-                    this.project.root,
-                    document.annotationFile
-                );
-                writers.set(change.uri, writer);
-            }
-            const changeHandler = this[change.type] as unknown as ChangeHandlerFunction<AnnotationFileChange>;
 
-            changeHandler(writer, document, change);
+        if (this.writeSapAnnotations) {
+            this.handleSapAnnotations(writers, changes);
+        } else {
+            for (const change of changes) {
+                let writer = writers.get(change.uri);
+                const document = this.documents.get(change.uri);
+                const cachedFile = this.fileCache?.get(change.uri);
+                if (!document || cachedFile === undefined || !this.facade) {
+                    continue;
+                }
+                if (!writer) {
+                    //writable cds document (augment)
+                    const textDocument = TextDocument.create(change.uri, 'cds', 0, cachedFile);
+                    writer = new CDSWriter(
+                        this.facade,
+                        this.vocabularyService,
+                        document.ast,
+                        document.comments,
+                        document.tokens,
```

Comment:
probably can pass document itself

Reply:
Yes, but that would be refactoring of already existing code. I'd prefer to keep that out of this PR and make a separate change for it.